### PR TITLE
Update timeseries docs

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/time_series.py [hunk 1]
```diff
@@ -48,7 +49,7 @@ class TimeSeriesCore(WriteableCogniteResource["TimeSeriesWrite"], ABC):
         description (str | None): Description of the time series.
         security_categories (Sequence[int] | None): The required security categories to access this time series.
         data_set_id (int | None): The dataSet ID for the item.
-        legacy_name (str | None): Set a value for legacyName to allow applications using API v0.3, v04, v05, and v0.6 to access this time series. The legacy name is the human-readable name for the time series and is mapped to the name field used in API versions 0.3-0.6. The legacyName field value must be unique, and setting this value to an already existing value will return an error. We recommend that you set this field to the same value as externalId.
+        legacy_name (str | None): : This field is not used by the API and will be removed in a future version of the SDK.
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:52]: What about `October 2024` instead? To match when we'll do a few updates at the same time as dropping PY38 support?

File: cognite/client/data_classes/time_series.py [hunk 2]
```diff
@@ -100,7 +108,7 @@ class TimeSeries(TimeSeriesCore):
         data_set_id (int | None): The dataSet ID for the item.
         created_time (int | None): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
         last_updated_time (int | None): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
-        legacy_name (str | None): Set a value for legacyName to allow applications using API v0.3, v04, v05, and v0.6 to access this time series. The legacy name is the human-readable name for the time series and is mapped to the name field used in API versions 0.3-0.6. The legacyName field value must be unique, and setting this value to an already existing value will return an error. We recommend that you set this field to the same value as externalId.
+        legacy_name (str | None): No description.
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:111]: Missing description

File: cognite/client/data_classes/time_series.py [hunk 3]
```diff
@@ -242,7 +250,7 @@ class TimeSeriesWrite(TimeSeriesCore):
         description (str | None): Description of the time series.
         security_categories (Sequence[int] | None): The required security categories to access this time series.
         data_set_id (int | None): The dataSet ID for the item.
-        legacy_name (str | None): Set a value for legacyName to allow applications using API v0.3, v04, v05, and v0.6 to access this time series. The legacy name is the human-readable name for the time series and is mapped to the name field used in API versions 0.3-0.6. The legacyName field value must be unique, and setting this value to an already existing value will return an error. We recommend that you set this field to the same value as externalId.
+        legacy_name (str | None): : This field is not used by the API and will be removed in a future version of the SDK
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:253]: I suggest we don't add this to `TimeSeriesWrite` as it was introduced without it.

File: cognite/client/data_classes/time_series.py (hunk 1 above)
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:52]: Note the double colon:
```suggestion
        legacy_name (str | None): This field is not used by the API and will be removed in a future version of the SDK.
```

File: cognite/client/data_classes/time_series.py (hunk 3 above)
Comment (doctrino) [reply cognite/client/data_classes/time_series.py:253]: It is currently in `TimeSeriesWrite`, unfortunately. https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/data_classes/time_series.py#L245

Comment (haakonvt) [reply cognite/client/data_classes/time_series.py]: Ah, my bad

File: cognite/client/data_classes/time_series.py [hunk 4]
```diff
@@ -77,6 +78,13 @@ def __init__(
         self.description = description
         self.security_categories = security_categories
         self.data_set_id = data_set_id
+        if legacy_name is not None:
+            warnings.warn(
+                "The 'legacy_name' is not used and will be ignored. It will be completely removed "
+                "in the next major version.",
```
Comment (haakonvt) [inline cognite/client/data_classes/time_series.py:84]: This should reflect the calendar deprecation date (October 2024)

**Review by haakonvt — APPROVED** [review]
One last comment

**Pull request merged**